### PR TITLE
rename lock free queue and remove wait_for_event API

### PR DESCRIPTION
**Pull request by irobot-ros**
Signed-off-by: Alberto Soragna <alberto.soragna@gmail.com>

### CODE REVIEW

File: rclcpp/include/rclcpp/experimental/buffers/lock_free_events_queue.hpp [hunk 1]
```diff
@@ -38,11 +38,11 @@ namespace buffers
  * queue aims to fix the issue of publishers being blocked by the executor extracting
  * events from the queue in a different thread, causing expensive mutex contention.
  */
-class BlockingConcurrentQueue : public EventsQueue
+class LockFreeEventsQueue : public EventsQueue
```
Comment (mauropasse) [inline rclcpp/include/rclcpp/experimental/buffers/lock_free_events_queue.hpp:41]: As the name has changed you should change 
`RCLCPP__EXPERIMENTAL__BUFFERS__BLOCKING_CONCURRENT_QUEUE_HPP_` with a new name

File: rclcpp/include/rclcpp/experimental/buffers/lock_free_events_queue.hpp [hunk 2]
```diff
@@ -107,26 +107,20 @@ class BlockingConcurrentQueue : public EventsQueue
...
-
-    // If no timeout specified, just wait for an event to arrive
-    event_queue_.wait_dequeue(event);
-    return true;
+    return event_queue_.wait_dequeue_timed(event, timeout);
   }
 
 private:
-  moodycamel::BlockingConcurrentQueue<rclcpp::executors::ExecutorEvent> event_queue_;
+  moodycamel::LockFreeEventsQueue<rclcpp::executors::ExecutorEvent> event_queue_;
```
Comment (mauropasse) [inline rclcpp/include/rclcpp/experimental/buffers/lock_free_events_queue.hpp:123]: Does this work? it comes from the examples of the queue:
`moodycamel::BlockingConcurrentQueue<int> q;` in https://github.com/cameron314/concurrentqueue

Comment (irobot-ros) [reply rclcpp/include/rclcpp/experimental/buffers/lock_free_events_queue.hpp:123]: Stupid replace all

**Comment by mauropasse** [comment]
If all the spin versions work fine with these changes I think it can be merged. Now we're blocking at dequeue, does any version of spin requires not to block?

**Comment by irobot-ros** [comment]
There is always a check to `size()` or a timeout when calling dequeue.
Indeed one of the problems of the previous version was that
```
        std::unique_lock&lt;std::mutex&gt; lock(this-&gt;push_mutex_);
        rclcpp::executors::ExecutorEvent event = event_queue_.front();
        event_queue_.pop();
        return event;
```

this would cause a segmentation fault if there are no events in the queue, so the API implicitly was requiring to check before calling it.

**Pull request merged**